rs/p2p/quic_transport/src/connection_handle.rs

```diff
@@ -50,6 +54,64 @@ pub struct ConnectionHandle {
     conn_id: ConnId,
 }
 
+fn observe_conn_error(err: &ConnectionError, op: &str, metrics: &QuicTransportMetrics) {
+    match err {
+        // TODO: most likely this can be made infallible
+        ConnectionError::LocallyClosed => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_LOCALLY_CLOSED_CONN])
+            .inc(),
+        ConnectionError::ApplicationClosed(_) => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_APP_CLOSED_CONN])
+            .inc(),
+        _ => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_INTERNALLY_CLOSED_CONN])
```

Comment:
This label can be misleading.  For example [`ConnectionError::Reset`](https://docs.rs/quinn/latest/quinn/enum.ConnectionError.html#variant.Reset) is not an internally closed connection?

Also might be good to have a comment on what the difference between `internally` and `locally` closed connections is.

Reply:
(I'm not sure I understand the difference myself from the names)

Reply:
the intent of internally is to specify connection closures triggered by the quic protocol and not by our code. I will expand it.